Added knowledge base to docs repo (#708)
128technology/docs: kb/2024-04-25-I95-56437.md

---
title: Application Identification incompatible with DSCP Steering
slug: I95-56437
date: 2024-4-25
tags: ['5.6', '5.6.13']
hide_table_of_contents: false
---

<!-- truncate -->

**Issue ID:** I95-56437  
**Last Updated:** 2024-04-25  
**Introduced in SSR Version:** 5.6.13

### Problem
The app-id and dscp-steering features are incompatible.  

If app-id is configured on a child service (application-name, domain-name, domain-name-category, url, subcategory) and also on the same service a dscp-range list is defined, this will trigger a failure state on config ingestion that causes the service to be added/modified/removed twice during config processing (ServiceManager + AppIdManager).  This causes config ingestion to fail and any subsequent service config updates *within that failed config* are ignored.  

The adminstrator will not be aware of a configuration issue. There is just a log to indicate this failure.  
`highway.log`
```
Apr 17 13:59:54.622 [HWMC|SVC ] WARN  (routingAgPoller) ServiceAddException: Failed to Add Service Audio.MicrosoftTeams
Apr 17 13:59:54.622 [HWMC|SVC ] ERROR (routingAgPoller) ServiceManager failed to process config trackers
```

### Release Notes

> The app-id and dscp-steering features are incompatible and should not be configured simultanesouly; doing so will cause future commits to fail. Added strict configuration validation to prevent configuring both features simultanesouly.

Reviewer: ```suggestion
The app-id and dscp-steering features are incompatible and should not be configured simultanesouly; doing so will cause future commits to fail. Added strict configuration validation to prevent configuring both features simultaneously.
```

> The app-id and dscp-steering features are incompatible and should not be configured simultaneously; doing so will cause future commits to fail. Added strict configuration validation to prevent configuring both features simultaneously.

Reviewer: ```suggestion
The app-id and dscp-steering features are incompatible and should not be configured simultaneously; doing so will cause future commits to fail. Strict configuration validation has been added to prevent configuring both features simultaneously.
```

### Severity
<details>
The potential impact of a software defect if encountered. Severity levels are:
* Critical: Could severely affect service, capacity/traffic, and maintenance capabilities. May have a prolonged impact to the entire system.
* Major: Could seriously affect system operation, maintenance, administration and related tasks.

> * Minor: Would not significantly impair the functioning or or affect service.

Reviewer: ```suggestion
* Minor: Would not significantly impair the functioning or affect service.
```

</details>
Major

### Status
Open

### Resolved In

### Product
SSR

### Functional Area
Configuration

### Workaround
<details>
Juniper may provide a method to temporarily circumvent a problem; workarounds do not exist for all issues.
</details>

> Disabling the service or deleting the offending config works (pick one feature or the other).

Reviewer: ```suggestion
Disable the service, or delete the offending configuration.
```
